scripts/cluster.py

```diff
@@ -0,0 +1,57 @@
+# This script can be used to start Ray on an existing cluster.
+#
+# How to use it: Create a file "nodes.txt" that contains a list of the IP
+# addresses of the nodes in the cluster. Put the head node first. This node will
+# host the driver and the scheduler.
+
+import argparse
+import paramiko
+import IPython
+
+parser = argparse.ArgumentParser(description='Parse information about the cluster.')
+parser.add_argument("--nodes", default="nodes.txt", help="Test file with node IP addresses, one line per address.")
+parser.add_argument("--key-file", type=str, help="Path to the file that contains the private key.")
+parser.add_argument("--username", default="ubuntu", help="User name for logging in.")
+
+def connect(hostname, username, key):
+  c = paramiko.SSHClient()
+  c.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+  print "connecting to ", hostname
+  c.connect(hostname=hostname, username="ubuntu", pkey=key)
+  return c
+
+def install_ray_with(nodes, username, key):
+  connections = map(lambda hostname: connect(hostname, username, key), nodes)
+  for c in connections:
+    c.exec_command("sudo apt-get update; sudo apt-get -y install git; git clone https://github.com/amplab/ray; cd ray; bash setup.sh")
+
+def start_ray_with(nodes, username, key, worker_path):
```

Comment:
If the nodes in the cluster try to prompt the user for the sudo password, that prompt probably won't be propagated back to the person who launched this script.